components/ulp/ulp_riscv/ulp_core/ulp_riscv_print.c

```diff
@@ -49,3 +49,24 @@ void ulp_riscv_print_hex(int h)
         h <<= 4; // move the 2nd leftmost byte to the left, to be extracted next
     }
 }
+
+void ulp_riscv_print_hex_with_number_of_digits(int h, int number_of_digits)
```

Comment:
Since the input here is `int h`, the maximum number of hex digits would be 8. In this case, I assume the purpose of this function would be to print < 8 bytes. Is the understanding correct?